examples/repeat_until_success.md

```diff
@@ -0,0 +1,477 @@
+grid_samples = SAMPLE_RATE*GRID_SPACING
+
+# Number of clock cycles needed for evaluating while condition (found empirically)
+LOOP_OVERHEAD = 8   
+# Duration of one iteration, expressed in number samples
+iteration_len_samples = (feedback_latency + LOOP_OVERHEAD)*CLOCK_TO_SAMPLE_SG
+
+# Increase the length of one cycle to reach the closest integer multiple of `grid_samples`
+cycle_len_samples = int(np.ceil(iteration_len_samples/grid_samples)*grid_samples)
+# Express the duration in clock cycles again (needed in the seqc program)
+cycle_len_clocks = iteration_len_samples//CLOCK_TO_SAMPLE_SG
+
+# Subtract the length of the `try` waveform to get the length of the `startQA` section
+section_len = iteration_len_samples - waveform_try_len
+
+cycle_len_samples/SAMPLE_RATE
+```
+
+## SHFQA sequencer program
+
+
+Now instruct the SHFQA on what it should do and with which timing, by writing the instructions in a sequencer program. In particular, we tell the SHFQA to:
+1. Stay silent for `waveform_try_len` samples, while the SHFSG plays the "try" waveform;
+2. Start a section with a duration `section_len`, previously calculated such that `section_len + waveform_try_len` is an integer multiple of 200 ns;
+3. During this section, perform the readout with `startQA` and forward the result of the readout to the PQSC readout register bank address specified by `READOUT_REGISTER`;
+4. Update the feedback time for the next iteration.
+These steps are repeated for `SIMULATED_FAILURES` times simulating a readout with result 0. Then we simulate an additional readout with result 1.
+
+<u>Note 1</u>:
+Instructions about the readout are provided through the arguments of the function `startQA()`; for this experiment the following parameters are particularly relevant: 
+- the first argument of `startQA()` is in the form `QA_GEN_i`, where `i` is the index of the waveform in the waveform memory to be played;
+- the second argument of `startQA()` is in the form `QA_INT_i`, where `i` is the index of the weight in the integration weights memory to be used during integration;
+- the fourth argument of `startQA()` is the address of the PQSC readout register where the result of the readout is sent.
+
+<u>Note 2</u>: the way to force the duration of the readout section to be `section_len` is to execute a `playZero(section_len)` before `startQA`. This may seem confusing at first, but the reason why this works is that `playZero` can run in parallel with a `startQA` and will only block other `playZero`. In practice, this will force the duration of the readout section to be `section_len`.
+
+```python
+READOUT_REGISTER = 0 # Address of PQSC readout register
+
+seqc_program_shfqa = Sequence()
+
+# Assign constants
+constants = {
+    "feedback_latency": feedback_latency,
+    "SIMULATED_FAILURES": SIMULATED_FAILURES,
+    "cycle_len_clocks": cycle_len_clocks,
+    "READOUT_REGISTER": READOUT_REGISTER,
+    "waveform_try_len": waveform_try_len,
+    "section_len": section_len,
```

Comment:
Is there a reason why some of them are not capital? I would unify it if there is no intention behind it.

Reply:
Some variables are constants (e.g. the readout register), others are the result of previous calculations